tests/unit_tests/test_tethys_services/test_backends/test_arcgis_portal.py

```diff
@@ -0,0 +1,101 @@
+from unittest import mock
+from django import test
+from tethys_services.backends.arcgis_portal import ArcGISPortalOAuth2
+
+
+@test.override_settings(
+    OAUTH_CONFIG={
+        'SOCIAL_AUTH_ARCGIS_PORTAL_URL': 'https://my.test.arcgis/portal'
+    }
+)
```

Comment:
I took a look at why your tests are failing. I'm guessing that the `ArcGISPortalOAuth2` class and class level logic is processed before this decorator is applied, causing the error. Generally, logic at the class level causes issues when testing. Is there a method on the class that you can use to do your check for the portal url?

Reply:
The tests are now fixed, so you can give this one final review. Thanks!